packages/core/src/vuePlugins/actions/useExtendedMasterportalapiMarkers/index.ts

```diff
@@ -44,6 +63,70 @@ export function useExtendedMasterportalapiMarkers(
         isVisible(feature) ? feature.getGeometry() : null
     })
 
+  const updateSelection = (feature: Feature | null) => {
+    if (feature === null) {
+      selected?.setStyle(undefined)
+      selected = null
+      return
+    }
+
+    let selectedCluster = feature.get('features') ? feature : null
+
+    if (selectedCluster === null) {
+      // @ts-expect-error | Really, it's there!
+      const candidates = map
+        .getLayers()
+        .getArray()
+        .find((layer) => layer.get('id') === feature.get('_gfiLayerId'))
+        // @ts-expect-error | Really, it's there!
+        .getSource()
+        .getFeaturesInExtent(
+          map.getView().calculateExtent(map.getSize()),
+          map.getView().getProjection()
+        )
+
+      selectedCluster = candidates.find((candidate) =>
+        candidate.get('features').includes(feature)
+      )
+    }
+
+    selectedCluster?.setStyle(
```

Comment:
Writing something like 

```ts
 const selectedCluster: Feature<Geometry> = feature.get('features')
      ? feature
      : // @ts-expect-error | Really, it's there!
        map
          .getLayers()
          .getArray()
          .find((layer) => layer.get('id') === feature.get('_gfiLayerId'))
          // @ts-expect-error | Really, it's there!
          .getSource()
          .getFeaturesInExtent(
            map.getView().calculateExtent(map.getSize()),
            map.getView().getProjection()
          )
          .find((candidate) => candidate.get('features').includes(feature))
```
Seems more fitting here, as the `selectedCluster` is always defined.

The addition of `: Feature<Geometry>` would be needed for the same reason as `candidates`.

Reply:
https://github.com/Dataport/polar/pull/43/commits/5fb1d59dbf3230b41c342572ba74871519b2bb6a

Oh, nice.